lib/intent-models/channel.js

```diff
@@ -0,0 +1,67 @@
+
+module.exports = {}; 
+const IntentBase = require('./base');
+// eslint-disable-next-line no-unused-vars
+const Schema = require('../models/schema');
+const { IntentMessage } = require('./message');
+const { IntentOperation } = require('./operation');
+const { IntentChannelParameter } = require('./channel-parameter');
+
+/**
+ * 
+ * @class
+ * @alias module:@asyncapi/parser#IntentChannel
+ * @extends IntentBase
+ */
+class IntentChannel extends IntentBase {
+  /**
+   * @returns {IntentOperation[]}
+   */
+  operations() { return [new IntentOperation()];}
+
+  /**
+   * @returns {string}
+   */
+  path() { return 'some/test/path'; }
+
+  /**
+   * @returns {IntentMessage[]}
+   */
+  messages() { return [new IntentMessage()];}
+  
+  /**
+   * 
+   * @param {string} bindingProtocol 
+   * @returns {boolean}
+   */
+  hasBinding(bindingProtocol) { return true; }
+  /**
+    * 
+    * @param {string} bindingProtocol 
+    * @returns {any|undefined} 
+    */
+  binding(bindingProtocol) {return {};}
```

Comment:
I would suggest we use enums here for `bindingProtocol` rather than `string`, what do you think?

Reply:
Question: are users able to create bindings without modifying the spec? If so, we should not use enum here since users may want to create their own bindings and then, will need to change the parser when in fact won't be needed.

Reply:
They cannot, bindings are part of spec, what you are thinking of is extensions. 

Binding fields are referred to as fixed fields: https://www.asyncapi.com/docs/specifications/2.0.0#channelBindingsObject

Reply:
Then yes, we can do it. I will update the doc introducing this change asap. Would you mind updating this mock if you have time? Otherwise I'll do.

Reply:
Hmmm, now that I think about it, what if we did:
```js
  /**
   * 
   * @param {string | ProtocolEnum} bindingProtocol 
   * @returns {boolean}
   */
  hasBinding(bindingProtocol) { return true; }
```
This way we don't restrict any use cases while still providing the same help for tooling if they want it? 🤔

Reply:
While enums, in general, make sense, I am not sure there does not exist a use case where strings are easier to use..

Reply:
This is just the detail of implementation. Each programming language implement (or not) enums in a different fashion. At the end an enum here means an `string` that can be **only** one value of a fixed list.

Reply:
alright, well I adapted the mock.